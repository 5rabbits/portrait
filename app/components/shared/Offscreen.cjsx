React = require('react')

@Offscreen = React.createClass
  render: ->
    return (
      <div className="offscreen-wrapper">
        <div className="offscreen-aside">
          { @props.panel }
        </div>
        <div className="offscreen-content">
          { @props.content }
        </div>
      </div>
    )

module.exports = @Offscreen
