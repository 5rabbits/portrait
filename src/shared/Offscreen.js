import React, { PureComponent } from 'react';

export default class Offscreen extends PureComponent {
  render() {
    return (
      <div className="offscreen-wrapper">
        <div className="offscreen-aside">
          { this.props.panel }
        </div>
        <div className="offscreen-content">
          { this.props.content }
        </div>
      </div>
    );
  }
}
