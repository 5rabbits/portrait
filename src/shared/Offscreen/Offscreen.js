import React from 'react'
import childrenPropType from 'propTypes/children'

const Offscreen = props =>
  <div className="offscreen-wrapper offscreen--show">
    <div className="offscreen-aside">
      {props.panel}
    </div>
    <div className="offscreen-content">
      {props.content}
    </div>
  </div>

Offscreen.propTypes = {
  content: childrenPropType,
  panel: childrenPropType,
}

export default Offscreen
