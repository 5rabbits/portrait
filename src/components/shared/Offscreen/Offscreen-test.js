import { React, toJSON, mount } from 'test/helper'
import Offscreen from 'shared/Offscreen'

describe('Offscreen', () => {
  it('should render with minimum props', () => {
    const component = mount(<Offscreen />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
