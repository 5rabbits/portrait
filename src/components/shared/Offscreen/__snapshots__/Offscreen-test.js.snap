// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offscreen should render with minimum props 1`] = `
<Offscreen>
  <div
    className="offscreen-wrapper offscreen--show"
  >
    <div
      className="offscreen-aside"
    />
    <div
      className="offscreen-content"
    />
  </div>
</Offscreen>
`;
